### ISSUE TITLE
Publicize: Add Support for New Facebook Pages

### ISSUE THREAD

**Issue by obenland** [Automattic/wp-calypso#51371]
Facebook is rolling out a new version of their Pages feature, changing the permissions that apps need to add content.

> For Pages that have been migrated to the New Page Experience, a Page access token is now required for all available endpoints
> —https://developers.facebook.com/docs/pages/overview/permissions-features 

There is some [more information about page access tokens](https://developers.facebook.com/docs/pages/access-tokens), but I think the solution is to adapt the code to get and store these tokens if they’re not available the next time we attempt to call the API on behalf of a given user. Ideally, we would do this in Keyring as an extension of the Facebook service.

We received the following tips from a Facebook contact, maybe we could start with that:
> You should use the (`/feed`)  [Pages Post endpoint](https://developers.facebook.com/docs/pages/publishing/) for performing the creation of posts, using the proper [`pages_manage_posts`](https://developers.facebook.com/docs/pages/overview-1#permissions) permission.
>
> If you have the information to troubleshoot (app ID /token) you can use the [Access Token debugger](https://developers.facebook.com/tools/debug/accesstoken/) to evaluate if your token is active/working, and you can debug the API calls using our [Graph API explorer](https://developers.facebook.com/tools/explorer/), calling to /feed (Pages Post) endpoint.
>
> In case the deprecated permissions aren't working, please test with the new one (`pages_manage_posts`)

For more context: p7fD6U-4lZ-p2

**Comment by JenSwish** [comment]
3818513-zen

**Comment by bobmatyas** [comment]
3809836-zen

**Comment by mikeicode** [comment]
Unable to connect to Facebook, stopped working when the page was changed to the new layout.

3873779-zd-woothemes

**Comment by StefMattana** [comment]
#3853988-zen

**Comment by desnum** [comment]
3877233-zen

**Comment by happychait** [comment]
3882885-zen

**Comment by ccwalburn** [comment]
Another case: 115623-hc

**Comment by happychait** [comment]
9258488-hc

**Comment by pablinos** [comment]
I've gone through some of these customer reports of the problem, and it seems it's actually related to the `can_post` field not being returned in the data we retrieve for pages that have been transitioned to the new experience.

I've contacted someone at Facebook to ask about the problem, and whether we can get access to a test page with the new experience enabled. In the meantime I'll try and update the code to not rely on this field.

**Comment by augurlu** [comment]
3911210 - zen

**Comment by StefMattana** [comment]
3934605-zen

**Comment by mikeicode** [comment]
15698640-hc please follow up when resolved 3960591-zd-woothemes

**Comment by StefMattana** [comment]
3955375-zen

**Comment by mikeicode** [comment]
> If we are able to contact any of the affected customers to check, that would be great. 

I just contacted the customer I mentioned earlier here asking them to try the connection, waiting on a response now. 3960591-zd-woothemes

**Comment by StefMattana** [comment]
Confirming that's working again for the user in ticket 3934605-zen :)

**Comment by pablinos** [comment]
This is certainly an odd one! I've checked the access token and it was created today, but there is no associated Publicize connection. This could be related to the new page experience, but it's presenting a bit differently to the problem that caused this issue.

> they even see a "Success" dialog

Reading how they describe this, I'm not sure it's the success dialog we would expect to see as the Publicize connection is created. The Facebook authentication process happens in a pop-up, and it can look like that has completed successfully but then the last test will fail.

The user access token looks valid for the Facebook page they're enquiring about too, but querying against it I see that the `can_post` field is coming back as `false`. Is it possible that the Facebook user they're logged in as, no longer has permission to post to that page?

There is the resulting page token which gives a `can_post` value of `true`. I'll do some more digging and see what I can find as the two `can_post` values should be the same.

**Comment by tmmbecker** [comment]
@pablinos I've been troubleshooting this same issue on https://thefifthestate.com.au/ and I have been able to reliably reproduce this problem as I direct access to the customer's Facebook account. I experienced the exact same problem that @jorpdesigns described. I get all of the [proper](https://d.pr/i/xOvbkd) [steps](https://d.pr/i/MMsaZd) in the flow, the user connection is created and I get a [success message](https://d.pr/i/fayW5C), but after the last step, I still [see the button to **Connect](https://d.pr/i/5B6l8K)**.

They are an admin on the page and the Page is part of the Meta Business Suite and has been updated to "the new Pages experience." They had previously connected the site to the old page but it appears that when they switched the page, it was automatically disconnected. They didn't do anything to disconnect it manually. 

I'm happy to share the details I've got if you'd like to do some more specific troubleshooting.

**Comment by eduardozulian** [comment]
I asked the customer that @jorpdesigns helped if they are an admin but given @tmmbecker's reply, it might be another issue.

@Automattic/dotcom-pa-group would you mind checking this report and see if any extra triaging (including setting a priority) needs to be done?

**Comment by jamiepalatnik** [comment]
Another report: 36037823-hc

**Comment by Robertght** [comment]
@pablinos any updates on this?

**Comment by mxhassani** [comment]
Similar issue here: 5384307-zd-woothemes

**Comment by dcoleonline** [comment]
- [ ] 5427285-zen

I told them to switch to classic pages for now https://web.facebook.com/business/help/664061057662340?_rdc=1&_rdr

**Comment by tmmbecker** [comment]
Update to my [previous comment](https://github.com/Automattic/wp-calypso/issues/51371#issuecomment-1179443081). The publisher switched back to classic pages (losing months of posts in the process) and was able to connect publicize with no trouble.

**Comment by tmmbecker** [comment]
@Automattic/dotcom-pa-group or @pablinos could someone please take a look?

**Comment by Robertght** [comment]
5464931-zen
